### PR TITLE
[GnuTLS] Upgrade to 3.7.8 (modified)

### PR DESCRIPTION
**Pull request by jeremiahpslewis**
Based on https://github.com/JuliaPackaging/Yggdrasil/pull/5670

### CODE REVIEW

**Comment by lucifer1004** [comment]
Nice fix!

File: G/GnuTLS/build_tarballs.jl [hunk 1]
```diff
@@ -29,25 +24,33 @@ if [[ ${target} == *darwin* ]]; then
     if [[ "${target}" == aarch64* ]]; then
         export CFLAGS="-mmacosx-version-min=11.0"
     else
-        export CFLAGS="-mmacosx-version-min=10.11"
+        export CFLAGS="-mmacosx-version-min=10.15"
```
Comment (giordano) [inline G/GnuTLS/build_tarballs.jl:27]: This is going to cut out many versions of macOS, is it really needed?

Comment (jeremiahpslewis) [reply G/GnuTLS/build_tarballs.jl]: 10.11 gives a Mac build error, trying 10.12

File: G/GnuTLS/build_tarballs.jl [hunk 2]
```diff
@@ -57,7 +59,7 @@ products = Product[
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency("Zlib_jll"),
-    Dependency("GMP_jll", v"6.2.0"),
+    Dependency("GMP_jll", v"6.2.1"),
     Dependency("Nettle_jll", v"3.7.2"; compat="~3.7.2"),
```
Comment (giordano) [inline G/GnuTLS/build_tarballs.jl:63]:
```suggestion
    Dependency("Nettle_jll"; compat="~3.7.2"),
```

**Pull request merged**